src/test/java/com/microsoft/azure/kusto/kafka/connect/sink/it/KustoSinkIT.java

```diff
@@ -0,0 +1,311 @@
+                        log.debug("JSON Record produced: {}", objectMapper.writeValueAsString(jsonRecordMap));
+                        producer.send(producerRecord);
+                    }
+                }
+                break;
+            case "csv":
+                producerProperties.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+                producerProperties.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+                // GenericRecords to json using avro
+                try (KafkaProducer<String, String> producer = new KafkaProducer<>(producerProperties)) {
+                    for (int i = 0; i < 10; i++) {
+                        GenericRecord record = (GenericRecord) randomDataBuilder.generate();
+                        Map<String, Object> jsonRecordMap = new TreeMap<>(record.getSchema().getFields().stream().parallel()
+                                .collect(Collectors.toMap(Schema.Field::name, field -> record.get(field.name()))));
+                        String objectsCommaSeparated = jsonRecordMap.values().stream().map(Object::toString).collect(Collectors.joining(","));
+                        log.debug("CSV Record produced: {}", objectsCommaSeparated);
+                        ProducerRecord<String, String> producerRecord = new ProducerRecord<>("e2e.csv.topic", "Key-" + i,
+                                objectsCommaSeparated);
+                        jsonRecordMap.put("type", dataFormat);
+                        expectedRecordsProduced.put(jsonRecordMap.get("vstr").toString(),
+                                objectMapper.writeValueAsString(jsonRecordMap));
+                        producer.send(producerRecord);
+                    }
+                }
+                break;
+        }
+        log.info("Produced messages for format {}", dataFormat);
+        Map<String, String> actualRecordsIngested = getRecordsIngested(dataFormat);
+        actualRecordsIngested.keySet().forEach(key -> {
+            log.debug("Record queried: {}", actualRecordsIngested.get(key));
+            try {
+                JSONAssert.assertEquals(expectedRecordsProduced.get(key), actualRecordsIngested.get(key),
+                        new CustomComparator(LENIENT,
+                                // there are sometimes round off errors in the double values but they are close enough to 8 precision
+                                new Customization("vdec", (vdec1,
+                                        vdec2) -> Math.abs(Double.parseDouble(vdec1.toString()) - Double.parseDouble(vdec2.toString())) < 0.000000001)));
+            } catch (JSONException e) {
+                fail(e);
+            }
+        });
+    }
+
+    private Map<String, String> getRecordsIngested(String dataFormat) {
+        // Waits 60 seconds for the records to be ingested
+        Predicate<Object> predicate = (results) -> results == null || ((Map<?, ?>) results).isEmpty();
```

Comment:
if only a few records were ingested ? you should actually wait on count

Reply:
I think E2E tests has this mechanism

Reply:
Added an assertion for the count as well